src/main/scala/viper/gobra/translator/implementations/translator/StatementsImpl.scala

```diff
@@ -37,6 +39,23 @@ class StatementsImpl extends Statements {
     def goA(a: in.Assertion): CodeWriter[vpr.Exp] = ctx.ass.translate(a)(ctx)
     def goE(e: in.Expr): CodeWriter[vpr.Exp] = ctx.expr.translate(e)(ctx)
 
+    /**
+      * Generates an Exhale statement for the assertions `pre` parametrized by `formalParams` and instantiated with `args`
+      */
+    def genExhale(pre: Vector[in.Assertion], formalParams: Vector[in.Parameter.In], args: Vector[in.Expr]): Writer[vpr.Stmt] = {
+      Violation.violation(args.length == formalParams.length, "number of passed arguments must match number of expected arguments")
+      val errorT = (x: Source.Verifier.Info, _: ErrorReason) => PreconditionError(x) dueTo GoCallPreconditionError(x)
+      for {
+        vArgss <- sequence(args map goE)
+        funcArgs <- sequence(formalParams map goE)
+        substitutions = (funcArgs zip vArgss).toMap
+        preCond <- sequence(pre map goA)
+        preCondInstance = preCond.map{ _.replace(substitutions) }
+        and = preCondInstance.foldRight[vpr.Exp](vpr.TrueLit()(pos, info, errT))((x,y) => vpr.And(x,y)(pos, info, errT))
```

Comment:
Please use `ViperUtil.bigAnd` since foldRight / foldLeft has significant performance impact on Silicon. Having a central place to change that (if ever necessary) seems advantageous to me

Reply:
I did not know about this function 😅 Thanks @ArquintL!